pkg/kudoctl/cmd/prompt/operator.go

```diff
@@ -0,0 +1,486 @@
+package prompt
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/Masterminds/semver"
+	"github.com/spf13/afero"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+	"github.com/kudobuilder/kudo/pkg/engine/task"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/cmd/generate"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages"
+)
+
+// ForOperator prompts to gather details for a new operator
+func ForOperator(fs afero.Fs, pathDefault string, overwrite bool, operatorDefault packages.OperatorFile) (*packages.OperatorFile, string, error) {
+
+	nameValid := func(input string) error {
+		if len(input) < 3 {
+			return errors.New("Operator name must have more than 3 characters")
+		}
+		return nil
+	}
+
+	name, err := WithValidator("Operator Name", operatorDefault.Name, nameValid)
+	if err != nil {
+		return nil, "", err
+	}
+
+	pathValid := func(input string) error {
+		if len(input) < 1 {
+			return errors.New("Operator directory must have more than 1 character")
+		}
+		return generate.CanGenerateOperator(fs, input, overwrite)
+	}
+
+	path, err := WithValidator("Operator directory", pathDefault, pathValid)
+	if err != nil {
+		return nil, "", err
+	}
+
+	versionValid := func(input string) error {
+		if len(input) < 1 {
+			return errors.New("Operator version is required in semver format")
+		}
+		_, err := semver.NewVersion(input)
+		return err
+	}
+	opVersion, err := WithValidator("Operator Version", operatorDefault.Version, versionValid)
+	if err != nil {
+		return nil, "", err
+	}
+
+	appVersion, err := WithDefault("Application Version", "")
+	if err != nil {
+		return nil, "", err
+	}
+
+	kudoVersion, err := WithDefault("Required KUDO Version", operatorDefault.KUDOVersion)
+	if err != nil {
+		return nil, "", err
+	}
+
+	url, err := WithDefault("Project URL", "")
+	if err != nil {
+		return nil, "", err
+	}
+
+	op := packages.OperatorFile{
+		Name:        name,
+		APIVersion:  operatorDefault.APIVersion,
+		Version:     opVersion,
+		AppVersion:  appVersion,
+		KUDOVersion: kudoVersion,
+		URL:         url,
+	}
+	return &op, path, nil
+}
+
+// ForMaintainer prompts to gather information to add an operator maintainer
+func ForMaintainer() (*v1beta1.Maintainer, error) {
```

Comment:
I wouldn't mind having this file split into more than one, but that's just a nit.